sdk/keyvault/azure-security-keyvault/azure/security/keyvault/aio/secrets/samples/hello_world_async.py

```diff
@@ -0,0 +1,101 @@
+
+# ----------------------------------------------------------------------------------------------------------
+# Prerequisites -
+#
+# 1. An Azure Key Vault-
+#    https://docs.microsoft.com/en-us/azure/key-vault/quick-create-cli
+#
+# 2. Microsoft Azure Key Vault PyPi package -
+#    https://pypi.python.org/pypi/azure-security-keyvault/
+#
+# 3. Set Environment variables AZURE_CLIENT_ID, AZURE_TENANT_ID, AZURE_CLIENT_SECRET
+#
+# 4. In the code, replace YOUR_VAULT_URL with your vault url.
+# ----------------------------------------------------------------------------------------------------------
+# Sample - demonstrates the basic CRUD operations on a vault(secret) resource for Azure Key Vault
+#
+# 1. Create a new Secret (set_secret)
+#
+# 2. Get an existing secret (get_secret)
+#
+# 3. Update an existing secret (set_secret)
+#
+# 4. Delete a secret (delete_secret)
+#
+# 5. Purge a secret(purge_deleted_secret)
+# ----------------------------------------------------------------------------------------------------------
+
+
+async def run_sample():
+    # Instantiate a secret client that will be used to call the service.
+    # Notice that the client is using default Azure credentials.
+    # To make default credentials work, ensure that environment variables 'AZURE_CLIENT_ID',
+    # 'AZURE_CLIENT_SECRET' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+    credential = AsyncDefaultAzureCredential()
+    client = SecretClient(vault_url="https://vault99e4122d.vault.azure.net", credential=credential)
+    try:
+        # Let's create a secret holding bank account credentials valid for 1 year.
+        # if the secret already exists in the key vault, then a new version of the secret is created.
+        print("\n1. Create Secret")
+        expires = datetime.datetime.now() + datetime.timedelta(days=365)
+        secret = await client.set_secret("secretName", "secretValue", expires=expires)
+        print("Secret with name '{0}' created with value {1}".format(secret.name, secret.value))
+        print("Secret with name '{0}' expires on {1}".format(secret.name, secret.expires))
+
+        # Let's get the bank secret using its name
+        print("\n2. Get a Secret by name")
+        bank_secret = await client.get_secret("secretName")
+        print("Secret with name '{0}' was found with value {1}.".format(bank_secret.name, bank_secret.value))
+
+        # After one year, the bank account is still active, we need to update the expiry time of the secret.
+        # The update method can be used to update the expiry attribute of the secret. It cannot be used to update
+        # the value of the secret.
+        print("\n3. Update a Secret by name")
+        expires = bank_secret.expires + datetime.timedelta(days=365)
+        updated_secret = await client.update_secret("secretName", expires=expires)
+        # To ensure secret is updated on the server side.
+        time.sleep(5)
```

Comment:
`time.sleep` in async code; also no need to sleep, none of the following code will fail if the secret isn't updated (and it will be when `update_secret` returns anyway).

Reply:
Since the next 2 lines print the updated_secret value, wanted to wait here to make sure we get the updated values.

Will update the `time.sleep` to asyncio.sleep.

Reply:
You have `updated_secret`, it has the updated values. There's no need to wait.